static/src/javascripts/projects/commercial/modules/dfp/create-slots.js

```diff
@@ -143,16 +143,30 @@ const adSlotDefinitions = {
   Note that for the DFP slot to be filled by GTP, you'll have to
   use addSlot from add-slot.js
 */
+
 const createAdSlotElements = (
     name: string,
     attrs: Object,
     classes: Array<string>
 ) => {
     const adSlots = [];
 
+    const id = `dfp-ad--${name}`;
+
+    // The next three lines are to prevent a problem that appeared with DCR
+    // by which an AdSlot with the given id might have already been
+    // statically introduced server side and needs to be removed before
+    // this code takes effect on the DOM.
+    // We are simply making sure that if we are about to introduce a DOM
+    // element with a given id, that we ensure there isn't already one.
+    const element = document.getElementById(id);
+    if (element && element.parentNode) {
+        element.parentNode.removeChild(element);
```

Comment:
Does this need to happen in fastdom?

Reply:
Maybe console.log an error there? It's good to defend against the duplicated ID situation, but if the code was sane, il shouldn't happen. A fat error in the log would prompt us to fix any misbehaving code.

Reply:
@jeteve Happy to add a `console.log`m but technically this is not sign of a problem per se, just frontend and dotcom-rendering having different logic :)

Reply:
@shtukas What happened to my query about fastdom? It seems my comment was just ignored.

Reply:
Discussed IRL :)